assembl/static/js/app/routeManager.js

```diff
@@ -408,19 +412,47 @@ var routeManager = Marionette.Object.extend({
           });
       });
     widgetPromise.then(function(widget){
-      var CurrentWidgetView;
+      if (!widget){
+        var error_text = "Requested vote widget could not be found.";
+        alert(error_text);
+        return;
+      }
+      var currentWidgetView;
       var Views = require('./views/tokenVoteSession.js');
+      var currentWidgetViewOptions = {
+        model: widget
+      };
       if ((arg) && (arg === 'result')){
-        CurrentWidgetView = Views.TokenVoteSessionResultModal
+        currentWidgetView = Views.TokenVoteSessionResultModal;
+      }
+      else {
+        if ( showInModal ){
+          currentWidgetView = Views.TokenVoteSessionModal;
+        }
+        else {
+          currentWidgetView = Views.TokenVoteSessionView;
+        }
+        currentWidgetViewOptions["showConfirmationInModal"] = showInModal;
+      }
+      var instanciatedView = new currentWidgetView(currentWidgetViewOptions);
+
+      if ( showInModal ){
+        Ctx.setCurrentModalView(instanciatedView);
+        Assembl.slider.show(instanciatedView);
       }
       else {
-        CurrentWidgetView = Views.TokenVoteSessionModal
+        Assembl.groupContainer.show(instanciatedView);
+        // some CSS hacks to clear default page
+        $("body").css("background", "white");
```

Comment:
Why not make a css class with these values, eg:
```
.token-vote-session-modal {
  body {
    background: $light-background
  }
  #header {
    // Remove it so and make sure you don't cause positional issues on downstream elements
    display: none,
    overflow: visible
  }
}
```

And instead of hard-coding the css here, add this class to the highest element possible in the tree. And You should remove this css class on closing of the view.

Reply:
this CSS would ont work, as body element is not into a .token-vote-session-modal element. Remember it is use-and-throw-away code that will very soon be completely replaced by v2 UI